File: output/StarCoder/codenet/C++/Java/s194793609.java
import java.util.Scanner;
import java.util.Vector;

public class s194793609 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        Vector<Long> a = new Vector<Long>(n);
        for (int i = 0; i < n; i++) {
            a.add(sc.nextLong());
        }

        Vector<Long> abs_a = new Vector<Long>(n);
        for (int i = 0; i < n; i++) {
            abs_a.add(Math.abs(a.get(i)));
        }

        long ans = 0;
        long min = abs_a.get(0);
        for (long e : abs_a) {
            ans += e;
            if (min > e) {
                min = e;
            }
        }
        int num_negative = 0;
        for (long e : a) {
            if (e < 0) {
                num_negative++;
            }
        }

        if (num_negative % 2) {
            ans -= min * 2;
        }

        System.out.println(ans);
    }
}

File: output/StarCoder/codenet/C++/Python/s194793609.py
n = int(input())
a = [int(x) for x in input().split()]

abs_a = [abs(x) for x in a]

ans = 0
min = abs_a[0]
for e in abs_a:
    ans += e
    if min > e:
        min = e

num_negative = 0
for e in a:
    if e < 0:
        num_negative += 1

if num_negative % 2:
    ans -= min * 2

print(ans)